conans/model/options.py

```diff
@@ -182,12 +184,19 @@ def __setitem__(self, item, value):
 
     def _set(self, item, value):
         # programmatic way to define values, for Conan codebase
+        important = item[-1] == "!"
+        item = item[:-1] if important else item
+
         current_value = self._data.get(item)
         if self._freeze and current_value.value is not None and current_value != value:
             raise ConanException(f"Incorrect attempt to modify option '{item}' "
                                  f"from '{current_value}' to '{value}'")
         self._ensure_exists(item)
-        self._data.setdefault(item, _PackageOption(item, None)).value = value
+        v = self._data.setdefault(item, _PackageOption(item, None))
+        new_value_important = important or (isinstance(value, _PackageOption) and value.important)
+        if new_value_important or not v.important:
```

Comment:
@memsharded Is it not always `v.important == False` in this condition?

Reply:
Not really, the value could have a previously assigned value, (``setdefault`` will recover that value if it was assigned previously, not create the default one) and if that value was important when it was assigned, the previous ``v.important==True``. This is what avoids previous non important values to be overwritten by downstream values, even if those are not important.